src/ecdsa/mod.rs

```diff
@@ -398,10 +398,8 @@ impl<C: Signing> Secp256k1<C> {
                     }
 
                     counter += 1;
-                    // From 1.32 can use `to_le_bytes` instead
-                    let le_counter = counter.to_le();
-                    let le_counter_bytes : [u8; 4] = mem::transmute(le_counter);
-                    for (i, b) in le_counter_bytes.iter().enumerate() {
+                    let le_counter = counter.to_le_bytes();
+                    for (i, b) in le_counter.iter().enumerate() {
                         extra_entropy[i] = *b;
                     }
```

Comment:
This whole chunk could be just `extra_entropy[..4].copy_from_slice(&counter.to_le_bytes);`

Also below (can't comment there) `cast` is probably better than `as`. `extra_entropy.as_ptr().cast::<ffi::types::c_void>();`

Reply:
I used both suggestions, I was unable to justify the `as` -> `cast` change in the commit log, can you explain why its better please, just for my learning?